File: sistema-nomina/src/types/usuario.ts
export interface Usuario {
  id: number;
  nombreCompleto: string;
  email: string;
  rol: 'Admin' | 'RRHH' | 'Contador' | 'Auditor' | 'Operador' | string;
  permisosJson: string;
  activo: boolean;
  ultimoAcceso?: string;
  fechaCreacion?: string;
}

export interface LoginRequest {
  email: string;
  password: string;
  rememberMe?: boolean;
}

export interface LoginResponse {
  id: number;
  nombreCompleto: string;
  email: string;
  rol: string;
  permisosJson: string;
  activo: boolean;
  tokenSesion?: string | null;
}

export interface CrearUsuarioRequest {
  nombreCompleto: string;
  email: string;
  password: string;
  rol: string;
  permisosJson?: string;
}

export interface ActualizarUsuarioRequest {
  nombreCompleto: string;
  email: string;
  password?: string;
  rol: string;
  permisosJson: string;
  activo: boolean;
}

File: backend/Models/Usuario.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    [Table("Usuarios")]
    public class Usuario
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        [MaxLength(150)]
        public string NombreCompleto { get; set; } = string.Empty;

        [Required]
        [MaxLength(150)]
        [EmailAddress]
        public string Email { get; set; } = string.Empty;

        [Required]
        [MaxLength(255)]
        public string PasswordHash { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string Rol { get; set; } = "Operador";

        public string PermisosJson { get; set; } = "[]";

        public bool Activo { get; set; } = true;

        public DateTime? UltimoAcceso { get; set; }

        public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;

        public DateTime FechaActualizacion { get; set; } = DateTime.UtcNow;
    }
}